omnibus/files/private-chef-cookbooks/private-chef/recipes/nginx.rb

```diff
@@ -169,6 +169,25 @@
   end
 end
 
+if node['private_chef']['opscode-erchef']['stats_user']
+  stats_api_passwd = PrivateChef.credentials.get('opscode_erchef', 'stats_password')
+  stats_passwd_file = node['private_chef']['opscode-erchef']['stats_password_file']
+
+  execute 'stats_api_append_password' do
+    command "openssl passwd -apr1 '#{stats_api_passwd}' >> #{stats_passwd_file}"
+    action :nothing
+    sensitive true
+  end
```

Comment:
We need to store this password back into the secrets store somewhere.  Perhaps rather than doing this as an execute we can move this into a library and then generate the password when we generate the other passwords in the initial configuration parsing.

Reply:
Wait I get this password from the secret store (unless I don't understand how this works).  I do chef-server-ctl add-secret opscode_erchef stats_password "12345".  This derived password is just an scrypt salted hash of that password.

Reply:
I refactored this so that it autogenerates a password if not provided and also just generates the file in one go (instead of 2 steps).  This way the nginx server can get restarted properly only when the password file actually changes.